### PR TITLE
Upgrade HttpClient version to 4.5.5

### PR DESCRIPTION
**Pull request by bmuschko**
### Context

Upgrades to the latest version of HttpClient to fix a security vulnerability.

### CODE REVIEW

File: subprojects/docs/src/docs/release/notes.md [hunk 1]
```diff
@@ -197,6 +197,10 @@ The `java-base` plugin will now add an `<sourceSetName>AnnotationProcessor` conf
 The deprecated property `StartParameter.taskOutputCacheEnabled` has been removed.
 Use `StartParameter.buildCacheEnabled` instead.
 
+### HttpClient library upgraded to version 4.5.5
+
+Gradle has been upgraded to embed HttpClient version 4.5.5 over 4.4.1.
```
Comment (big-guy) [inline subprojects/docs/src/docs/release/notes.md]: Can you add a link to the release notes for HttpClient? https://archive.apache.org/dist/httpcomponents/httpclient/RELEASE_NOTES-4.5.x.txt

File: subprojects/docs/docs.gradle [hunk 2]
```diff
@@ -75,7 +75,7 @@ dependencies {
         exclude group: 'org.apache.httpcomponents'
     }
     testCompile libraries.commons_httpclient
-    testCompile 'org.apache.httpcomponents:httpmime:4.4.1'
+    testCompile 'org.apache.httpcomponents:httpmime:4.5.5'
```
Comment (big-guy) [inline subprojects/docs/docs.gradle]: `"org.apache.httpcomponents:httpmime:${versions.httpclient}"` to keep them in sync or do they just happen to have the same version now?

Comment (bmuschko) [reply subprojects/docs/docs.gradle]: This is just for running tests in the `docs` module. I just upgraded the version to keep them in sync.

File: subprojects/docs/src/docs/release/notes.md (hunk 1 above)
Comment (bmuschko) [reply subprojects/docs/src/docs/release/notes.md]: done

File: subprojects/docs/docs.gradle (hunk 2 above)
Comment (bmuschko) [reply subprojects/docs/docs.gradle]: Used the extra property.

**Pull request merged**